src/test/java/org/elasticsearch/rest/action/admin/indices/upgrade/UpgradeTest.java

```diff
@@ -142,32 +142,41 @@ public boolean apply(Object o) {
         logger.info("--> Nodes upgrade complete");
         logSegmentsState();
         
-        final HttpRequestBuilder httpClient = httpClient();
-
-        assertNotUpgraded(httpClient, null);
+        assertNotUpgraded(httpClient(), null);
         final String indexToUpgrade = "test" + randomInt(numIndexes - 1);
         
+        // #10213:
+        logger.info("--> Running upgrade on index " + indexToUpgrade + ", only ancient segments");
+        runUpgrade(httpClient(), indexToUpgrade, "only_ancient_segments", "true");
+
+        // TODO: why do we never test upgrading ancient segments here...?
+        // E.g. 0.90.x (Lucene 3.x) --> 1.x (Lucene 4.x)  ElasticsearchBackwardsCompatIntegrationTest.backwardsCompatibilityPath checks that
```

Comment:
Hrm, 0.90.x was 4.x right? The backcompat tests can only run on the same major version because of wire format changes between 1.x and 2.0.  I think the only way to test this is with the static bwc tests.  There used to be an "UpgradeReallyOldIndexTest" or something like that. Perhaps we should have that to test the ancient vs non ancient behavior, and leave the static bwc tests just doing a normal upgrade?